backend/src/test/java/com/github/esgoet/backend/services/BookServiceUnitTest.java

```diff
@@ -96,4 +97,52 @@ void deleteBook_Test() {
         bookService.deleteBook("1");
         verify(bookRepo).deleteById("1");
     }
+
+    @Test
+    void testUpdateBook_Success() {
+
+        // Given
+        String id = "1";
+        Book existingBook = new Book(id, "author1", "title1", Genre.THRILLER, "description1", "12345678", "cover1", localDate);
+        BookDto updatedBook = new BookDto("author2", "title2", Genre.FICTION, "description2", "23456789", "cover2", localDate);
+
+        // When
+        when(bookRepo.findById(id)).thenReturn(Optional.of(existingBook));
+        when(bookRepo.save(any(Book.class))).thenAnswer(invocation -> invocation.getArguments()[0]);
```

Comment:
would simplify it and just also have an updatedBook of type Book (and rename updatedBook to updatedBookDto)
```suggestion
        when(bookRepo.save(updatedBook).thenAnswer(updatedBook);
```